src/groundcrew/agent.py

```diff
@@ -250,7 +251,11 @@ def dispatch(self, user_prompt: str) -> None:
             self.spinner.start()
 
             # Run Tool
-            tool_response = self.run_tool(parsed_select_tool_response)
+            try:
```

Comment:
Trying to ensure the spinner is stopped if the tool fails...this might have been part of the problem